ledger/complete/wal/checkpoint_v6_reader.go

```diff
@@ -258,16 +257,31 @@ func readTopLevelTries(dir string, fileName string, subtrieNodes [][]*node.Node,
 	topLevelNodes := make([]*node.Node, topLevelNodesCount+1) //+1 for 0 index meaning nil
 	tries := make([]*trie.MTrie, triesCount)
 
-	totalSubTrieNodeCount := computeTotalSubTrieNodeCount(subtrieNodes)
-	// TODO: read subtrie Node count and validate
-
 	_, err = file.Seek(0, io.SeekStart)
 	if err != nil {
 		return nil, fmt.Errorf("could not seek to 0: %w", err)
 	}
 
 	reader := NewCRC32Reader(bufio.NewReaderSize(file, defaultBufioReadSize))
 
+	// read subtrie count
+	buf := make([]byte, encNodeCountSize)
+	_, err = io.ReadFull(reader, buf)
+	if err != nil {
+		return nil, fmt.Errorf("could not read subtrie node count: %w", err)
+	}
+	readSubtrieNodeCount, err := decodeSubtrieFooter(buf)
+	if err != nil {
+		return nil, fmt.Errorf("could not decode node count: %w", err)
+	}
```

Comment:
This error checking isn't needed if we decode subtrie node count directly above.